ctc_lib/container_tool_command.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ctc_lib
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleCloudPlatform/runtimes-common/ctc_lib/help"
+	"github.com/GoogleCloudPlatform/runtimes-common/ctc_lib/version"
+	"github.com/GoogleCloudPlatform/runtimes-common/ctc_lib/writer"
+	"github.com/spf13/cobra"
+)
+
+type ContainerToolCommand struct {
+	*cobra.Command
+	Phase                string
+	Version              string
+	HelpText             string
+	Output               interface{}
+	subCommandsOutputMap map[string]interface{}
+	args                 []string
+}
+
+type ContainerToolCommandList struct {
+	*cobra.Command
+	Phase                string
+	Version              string
+	OutputList           []interface{}
+	subCommandsOutputMap map[string]interface{}
+	args                 []string
+}
+
+// Initializes the Root Command with Common subcommands and Configs.
+func (ctc *ContainerToolCommand) init() {
+	// Add version subcommand
+	ctc.AddCommand(version.NewVersionCommand(ctc.Version,
+		ctc.Command.Name()).Command, &version.VersionOutput{})
+
+	// Add help subcommand
+	ctc.Command.SetHelpCommand(help.NewHelpCommand(ctc.HelpText,
```

Comment:
Does cobra do this automatically? Why do we need to do it manually here? Are we overriding the behavior with something different?

Reply:
no cobra does not do this automatically. Its the responsibility of the user to do it. If its not set, it defaults to Usage.